include/z64save.h

```diff
@@ -42,142 +42,193 @@ typedef struct {
+    /* 0x10 */ u32 collectible;
+} CycleSceneFlags; // size = 0x14
+
+typedef struct {
+    /* 0x00 */ u16 optionId;                     // "option_id"
+    /* 0x02 */ u8 language;                      // "j_n"
+    /* 0x03 */ s8 audioSetting;                  // "s_sound"
+    /* 0x04 */ u8 languageSetting;               // "language"
+    /* 0x05 */ u8 zTargetSetting;                // 0: Switch; 1: Hold
+} SaveOptions; // size = 0x06
+
+typedef struct {
+    /* 0x0000 */ char newf[6];                   // "newf"               Will always be "ZELDA3 for a valid save
+    /* 0x0006 */ u16 deaths;                     // "savect"
+    /* 0x0008 */ char playerName[8];             // "player_name"
+    /* 0x0010 */ s16 healthCapacity;             // "max_life"
+    /* 0x0012 */ s16 health;                     // "now_life"
+    /* 0x0014 */ s8 magicLevel;                  // "magic_max"
+    /* 0x0015 */ s8 magic;                       // "magic_now"
+    /* 0x0016 */ s16 rupees;                     // "lupy_count"
+    /* 0x0018 */ u16 swordHealth;                // "long_sword_hp"
+    /* 0x001A */ u16 tatlTimer;                  // "navi_timer"
+    /* 0x001C */ u8 magicAcquired;               // "magic_mode"
+    /* 0x001D */ u8 doubleMagic;                 // "magic_ability"
+    /* 0x001E */ u8 doubleDefense;               // "life_ability"
+    /* 0x001F */ u8 unk_1F;                      // "ocarina_round"
+    /* 0x0020 */ u8 unk_20;                      // "first_memory"
+    /* 0x0022 */ u16 owlActivationFlags;         // "memory_warp_point"
+    /* 0x0024 */ u8 unk_24;                      // "last_warp_pt"
+    /* 0x0026 */ s16 savedSceneNum;              // "scene_data_ID"
+} SavePlayerData; // size = 0x28
+
+typedef struct {
+    /* 0x0000 */ u32 entranceIndex;              // "scene_no"
+    /* 0x0004 */ u8 equippedMask;                // "player_mask"
+    /* 0x0005 */ u8 isFirstCycle;                // "opening_flag"
     /* 0x0006 */ u8 unk_06;
-    /* 0x0007 */ u8 linkAge;                 // "link_age"
-    /* 0x0008 */ s32 cutscene;               // "day_time"
-    /* 0x000C */ u16 time;                   // "zelda_time"
+    /* 0x0007 */ u8 linkAge;                     // "link_age"
+    /* 0x0008 */ s32 cutscene;                   // "day_time"
+    /* 0x000C */ u16 time;                       // "zelda_time"
     /* 0x000E */ u16 owlSaveLocation;
-    /* 0x0010 */ s32 isNight;                // "asahiru_fg"
-    /* 0x0014 */ u32 unk_14;                 // "change_zelda_time"
-    /* 0x0018 */ s32 day;                    // "totalday"
-    /* 0x001C */ u32 daysElapsed;            // "eventday"
-    /* 0x0020 */ u8 playerForm;              // "player_character"
-    /* 0x0021 */ u8 snowheadCleared;         // "spring_flag"
-    /* 0x0022 */ u8 hasTatl;                 // "bell_flag"
+    /* 0x0010 */ s32 isNight;                    // "asahiru_fg"
+    /* 0x0014 */ u32 daySpeed;                   // "change_zelda_time"
```

Comment:
Yeah. In kz it is named "time speed". It affects how fast the time goes. And a negative value makes the time go in reverse

Reply:
From z_parameter, it's clear that when the inverse song of time is played, `daySpeed` is set to `-2`, so I would say a negative day speed slows down time instead of making it go reverse:
https://github.com/engineer124/mm/blob/542bbfdeea2b4647ad82ad5b9b2c733a4be29b46/src/code/z_parameter.c#L2581
(This is the code that makes the diamond blue instead of green when time is slowed down)

Reply:
From KZ, -3 is "no flow" rather than 0

Reply:
I don't think any of these names are much good, though.

Reply:
`timeFlowRate` isn't quite it since that implies 0 is stopped. `timeFlowRateAdjustment` is better but a bit long. Anyone got any better ideas?

Reply:
I'm not a huge fan of neither of those names :s 
I think we could leave it name it like this until we have a better name for it, it isn't used much in the codebase